wandb/sdk/launch/runner/kubernetes_runner.py

```diff
@@ -219,6 +231,16 @@ def get_status(self) -> Status:
         self._delete_secret_if_completed(return_status.state)
         return return_status
 
+    def _is_container_creating(self, pod) -> bool:
+        return (
+            hasattr(pod.status, "container_statuses")
+            and pod.status.container_statuses
+            and hasattr(pod.status.container_statuses[0].state, "waiting")
+            and hasattr(pod.status.container_statuses[0].state.waiting, "reason")
+            and pod.status.container_statuses[0].state.waiting.reason
+            == "ContainerCreating"
```

Comment:
This is impressively compressed but I think we should really examine the full list of container statuses when we make this check rather than looking only at the 0 index. I'm not sure what this list of container statuses might look like in the wild but I have a feeling there are cases where this will not work

Reply:
Agreed. We should always check all statuses

Reply:
If it is instead some list of past statuses, we should ensure we comment and explain why we're checking a specific one